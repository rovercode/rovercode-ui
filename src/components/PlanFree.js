import React from 'react';
import { Grid, Typography } from '@material-ui/core';
import { hot } from 'react-hot-loader';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

const PlanFree = ({ freeSlots }) => (
  <Grid container direction="column" justify="center" alignItems="stretch">
    <Grid item>
      <Typography variant="subtitle2" align="center">
        { `${freeSlots} ` }
        <FormattedMessage
          id="app.plan_free.programs"
          description="Label for how many programs allowed"
          defaultMessage="program slots"
        />
      </Typography>
    </Grid>
    <Grid item>
      <Typography variant="subtitle2" align="center">
        <FormattedMessage
          id="app.plan_free.courses"
          description="Indicates a limited access to the courses feature"
          defaultMessage="Limited access to courses"
        />
      </Typography>
    </Grid>
  </Grid>
);

PlanFree.propTypes = {
  freeSlots: PropTypes.number.isRequired,
};

export default hot(module)(PlanFree);
