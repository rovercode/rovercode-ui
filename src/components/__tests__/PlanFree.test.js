import React from 'react';

import PlanFree from '../PlanFree';

describe('The PlanFree component', () => {
  test('renders on the page with no errors', () => {
    const wrapper = shallowWithIntl(<PlanFree freeSlots={2} />);
    expect(wrapper).toMatchSnapshot();
  });
});
