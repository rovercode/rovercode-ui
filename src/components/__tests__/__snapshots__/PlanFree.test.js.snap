// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PlanFree component renders on the page with no errors 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="stretch"
  container={true}
  direction="column"
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(Typography))
      align="center"
      variant="subtitle2"
    >
      2 
      <FormattedMessage
        defaultMessage="program slots"
        id="app.plan_free.programs"
      />
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(Typography))
      align="center"
      variant="subtitle2"
    >
      <FormattedMessage
        defaultMessage="Limited access to courses"
        id="app.plan_free.courses"
      />
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
